src/clj/rems/api/services/catalogue.clj

```diff
@@ -67,3 +67,32 @@
 
 (def get-localized-catalogue-items catalogue/get-localized-catalogue-items)
 (def get-localized-catalogue-item catalogue/get-localized-catalogue-item)
+
+(defn change-form!
+  "Changes the form of a catalogue item.
+
+  Since we don't want to modify the old item we must create
+  a new item that is the copy of the old item except for the changed form."
+  [item form-id]
+  ;; create a new item with the new form
+  (let [new-item (db/create-catalogue-item! {:enabled true
+                                             :archived false
+                                             :form form-id
+                                             :resid (:resource-id item)
+                                             :wfid (:wfid item)})]
+
+    ;; copy localizations
+    (doseq [[langcode localization] (:localizations item)]
+      (db/upsert-catalogue-item-localization! {:id (:id new-item)
+                                               :langcode (name langcode)
+                                               :title (:title localization)
+                                               :infourl (:infourl localization)}))
```

Comment:
could you use create-catalogue-item! from this ns instead of these db/ functions?

Reply:
That one creates a whole API response so I'd rather not use it internally. Duplicated is ok here for now. Maybe can DRY this later.